AutocompleteClient/FW/API/Parser/CIOResponseParser.swift

```diff
@@ -80,18 +80,22 @@ struct CIOResponseParser: AbstractResponseParser {
                             }
                             
                             if let groups = autocompleteResult.groups{
-                                for group in groups{
+                                let maximumNumberOfGroupItems = self.maximumNumberOfResultsForItem(result: autocompleteResult, at: index)
```

Comment:
I don't love `maximumNumberOfResultsForItem` as a function name.  

I suggest keeping `maximum` as the prefix and ending with `PerResult`, and figuring out something to stick in the middle referring to groups.

`maximumGroupsPerResult`, `maximumGroupsShownPerResult`, `maximumGroupResultsPerResult` ... something to that effect.

Reply:
Actually the name was `maximumNumberOfGroupsForItem` not `maximumNumberOfResultsForItem`, that was the adapter method implemented in the CIOAutocompleteViewController.

The delegate method that the client would have to implement is `optional func autocompleteController(controller: CIOAutocompleteViewController, maximumNumberOfGroupsForItem item: CIOAutocompleteResult, itemIndex: Int) -> Int`. 

But anyhow, I changed the name of that method to maximumGroupsShownPerResult.